library/init/core.lean

```diff
@@ -343,6 +343,8 @@ class has_ssubset  (α : Type u) := (ssubset : α → α → Prop)
    Example: {a, b, c}. -/
 class has_emptyc   (α : Type u) := (emptyc : α)
 class has_insert   (α : out_param $ Type u) (γ : Type v) := (insert : α → γ → γ)
+-- TODO: should I use `out_param` on one of the arguments?
```

Comment:
I think `α` should be an out-param, just like in `insert`.  I believe that otherwise `({42} : set ℝ)` won't work (because Lean can't figure out `42` should be a real).